src/components/drawer/drawer.ts

```diff
@@ -0,0 +1,168 @@
+interface Constructor {
+    el: any
+    location: string // Possible options: top, right, bottom, left
```

Comment:
Not sure if you want to, but you could maybe use an enum for this. I'm indifferent.

Reply:
oh for sure. you definitely could. Its not a big deal i can update it